File: python/utils/resolve_khinsider_url.py

import aiohttp
from bs4 import BeautifulSoup

async def resolve_khinsider_url(url):
    if not url.startswith('https://downloads.khinsider.com/game-soundtracks'):
        return url
    async with aiohttp.ClientSession() as session:
        async with session.get(url) as response:
            response.raise_for_status()
            content = await response.text()
            soup = BeautifulSoup(content, 'html.parser')
            audio_player = soup.find('audio')
            if audio_player:
                return audio_player['src']
    return url

File: typescript/utils/resolve_khinsider_url.ts
export async function resolveKhinsiderUrl(url: string): Promise<string> {
  try {
    const response = await fetch(url);
    const html = await response.text();
    const match = html.match(/<audio[^>]*src="([^"]+\.mp3)"[^>]*>/);
    if (match && match[1]) {
      return match[1];
    }
  } catch (error) {
    console.error(`Error resolving Khinsider URL ${url}: ${error.message}`);
  }
  return url; // Return original URL if resolution fails
}
